### ISSUE TITLE
Fine-tuned pytorch transformer model (for NER) packaged to spaCy model

### ISSUE THREAD

**Issue by arBest** [explosion/spacy-transformers#209]
Hi everyone,

The issue I have currently is that I have a pytorch transformer model trained to predict entities.
I basically build the model and package it with spacy and when I try to load it with `spacy.load('custom_transformer_model')` to get named entities (predicted by the transformer model) I get the following error:

```    
  nlp = spacy.load('en_biobert_v11')
  File "/lila/home/rajannaa/transformers/env/lib/python3.6/site-packages/spacy/__init__.py", line 30, in load
    return util.load_model(name, **overrides)
  File "/lila/home/rajannaa/transformers/env/lib/python3.6/site-packages/spacy/util.py", line 170, in load_model
    return load_model_from_package(name, **overrides)
  File "/lila/home/rajannaa/transformers/env/lib/python3.6/site-packages/spacy/util.py", line 191, in load_model_from_package
    return cls.load(**overrides)
  File "/lila/home/rajannaa/transformers/env/lib/python3.6/site-packages/en_biobert_v11/__init__.py", line 12, in load
    return load_model_from_init_py(__file__, **overrides)
  File "/lila/home/rajannaa/transformers/env/lib/python3.6/site-packages/spacy/util.py", line 239, in load_model_from_init_py
    return load_model_from_path(data_path, meta, **overrides)
  File "/lila/home/rajannaa/transformers/env/lib/python3.6/site-packages/spacy/util.py", line 222, in load_model_from_path
    return nlp.from_disk(model_path, exclude=disable)
  File "/lila/home/rajannaa/transformers/env/lib/python3.6/site-packages/spacy/language.py", line 974, in from_disk
    util.from_disk(path, deserializers, exclude)
  File "/lila/home/rajannaa/transformers/env/lib/python3.6/site-packages/spacy/util.py", line 690, in from_disk
    reader(path / key)
  File "/lila/home/rajannaa/transformers/env/lib/python3.6/site-packages/spacy/language.py", line 969, in <lambda>
    p, exclude=["vocab"]
  File "nn_parser.pyx", line 668, in spacy.syntax.nn_parser.Parser.from_disk
  File "nn_parser.pyx", line 56, in spacy.syntax.nn_parser.Parser.Model
TypeError: Model() takes exactly 1 positional argument (0 given)

```

The snippet to serialize and save model:

```
from spacy_transformers import TransformersLanguage, TransformersWordPiecer, TransformersTok2Vec
import os

name = "biobert"
results_dir= os.getcwd() + "/ner_outputs_BC5CDR-chem-IOB"
pytorch_path= results_dir + "/pytorch-biobert"

nlp = TransformersLanguage(trf_name=name, meta={"lang": "en"})
nlp.add_pipe(nlp.create_pipe("sentencizer"))
nlp.add_pipe(TransformersWordPiecer.from_pretrained(nlp.vocab, pytorch_path))
nlp.add_pipe(TransformersTok2Vec.from_pretrained(nlp.vocab, pytorch_path))
nlp.add_pipe(nlp.create_pipe("ner"), last=True)

print(nlp.pipe_names)

nlp.to_disk(results_dir + '/spacy')
```

The meta.json for building it looks like this:

```
{
  "lang":"en",
  "lang_factory":"trf",
  "name":"biobert_v11",
  "version":"0.0.0",
  "spacy_version":">=2.3.1",
  "description":"",
  "author":"",
  "email":"",
  "url":"",
  "license":"",
  "spacy_git_version":"554291558",
  "vectors":{
    "width":0,
    "vectors":0,
    "keys":0,
    "name":null
  },
  "pipeline":[
    "sentencizer",
    "trf_wordpiecer",
    "trf_tok2vec",
    "ner"
  ],
  "factories":{
    "sentencizer":"sentencizer",
    "trf_wordpiecer":"trf_wordpiecer",
    "trf_tok2vec":"trf_tok2vec",
    "ner":"ner"
  },
  "labels":{
    "ner":[
      "B-Disease-MISC",
      "I-Disease-MISC",
      "B-Chemical-MISC",
      "I-Chemical-MISC",
      "O-MISC"
    ]
  }
}
```

```
spaCy version    2.3.1
Platform         Linux-3.10.0-957.5.1.el7.x86_64-x86_64-with-centos-7.6.1810-Core
Python version   3.6.6
spacy_transformers version 0.6.2
```

I've been trying to go through spaCy docs, issues to understand what I'm doing wrong. Any help or pointing me to the right direction would be greatly appreciated.

Thanks in advance to the community!

**Comment by svlandeg** [comment]
Sorry for the late reply. If I understand your post correctly, you're creating a new NER component with `nlp.create_pipe("ner")`. While that creates a new pipeline component, you'd still need to initialize and train an underlying ML model (on top of the transformer output). This wasn't really supported with the previous versions of spacy-transformers.

The good news is however, that we recently released a `v1.0.0rc0` version of this library to work together with the revamped spaCy v3, for which a a pre-release is available as [`spacy-nightly`](https://pypi.org/project/spacy-nightly/). 

You can find general information on the new spaCy v3 here: https://nightly.spacy.io and transformer-specific documentation at https://nightly.spacy.io/usage/embeddings-transformers#transformers and https://nightly.spacy.io/api/transformer. 

In a nutshell, you'll find that the spaCy v3 provides much better and robust support for using transformers as part of your spaCy `nlp` pipelines, and the new config system gives you much more control to customize settings to your use-case. There is plenty of documentation, but let us know if you run into issues!
